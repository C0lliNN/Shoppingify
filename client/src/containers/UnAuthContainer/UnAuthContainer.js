import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';
import logo from '../../assets/images/logo.svg';

const Container = styled.div`
  width: 80%;
  max-width: 700px;
  margin: 20px auto 0px auto;
  @media (min-width: ${variables.MD_BREAK_POINT}px) {
    margin: 100px auto 0px auto;
  }
`;

const Card = styled.div`
  background-color: ${variables.COLORS.light_brown};
  border-radius: ${variables.BORDER_RADIUS_2};
  padding: 20px;
  box-sizing: border-box;
  box-shadow: 0px 0px 20px 0px #0000001c;
`;

const TitleBar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const Title = styled.h3`
  font-weight: 600;
  font-family: ${variables.FONT_FAMILY};
  font-size: ${variables.FONT_SIZE_4};
`;

function UnAuthContainer({ children, title }) {
  return (
    <Container>
      <Card>
        <TitleBar>
          <img src={logo} alt="Shoppingify" />
          <Title>{title}</Title>
          <img src={logo} alt="Shoppingify" />
        </TitleBar>
        {children}
      </Card>
    </Container>
  );
}

UnAuthContainer.propTypes = {
  children: PropTypes.any,
  title: PropTypes.string.isRequired,
};

export default UnAuthContainer;
