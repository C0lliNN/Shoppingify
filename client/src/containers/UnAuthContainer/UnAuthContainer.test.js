import React from 'react';
import { render } from '../../tests/utilities';
import UnAuthContainer from './UnAuthContainer';
import 'jest-styled-components';

const title = 'Container';
const childrenText = 'Custom Title';

function exec() {
  return render(
    <UnAuthContainer title={title}>{childrenText}</UnAuthContainer>
  );
}

describe('<UnAuthContainer/>', () => {
  it('should render correctly', () => {
    const { container } = exec();
    expect(container).toMatchSnapshot();
  });
  it('should display the title', () => {
    const { getByText } = exec();

    expect(getByText(title)).not.toBeNull();
  });
  it('should display the children', () => {
    const { getByText } = exec();

    expect(getByText(childrenText)).not.toBeNull();
  });
});
