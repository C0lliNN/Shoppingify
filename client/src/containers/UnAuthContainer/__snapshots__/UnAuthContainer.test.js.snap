// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnAuthContainer/> should render correctly 1`] = `
<div>
  <div
    class="sc-AxjAm kcPFpt"
  >
    <div
      class="sc-AxirZ fwsGNn"
    >
      <div
        class="sc-AxiKw gsASVG"
      >
        <img
          alt="Shoppingify"
          src="logo.svg"
        />
        <h3
          class="sc-AxhCb fnLSuL"
        >
          Container
        </h3>
        <img
          alt="Shoppingify"
          src="logo.svg"
        />
      </div>
      Custom Title
    </div>
  </div>
</div>
`;
